nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/DeleteHDFS.java

```diff
@@ -0,0 +1,161 @@
+            .name("success")
+            .description("FlowFiles will be routed here if the delete command was successful")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("FlowFiles will be routed here if the delete command was unsuccessful")
+            .build();
+
+    public static final PropertyDescriptor FILE_OR_DIRECTORY = new PropertyDescriptor.Builder()
+            .name("File or Directory")
+            .description("The HDFS file or directory to delete. A wildcard expression may be used to only delete certain files")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor RECURSIVE = new PropertyDescriptor.Builder()
+            .name("Recursive")
+            .description("Remove contents of a non-empty directory recursively")
+            .allowableValues("true", "false")
+            .required(true)
+            .defaultValue("true")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private static final Set<Relationship> relationships;
+
+    static {
+        final Set<Relationship> relationshipSet = new HashSet<>();
+        relationshipSet.add(REL_SUCCESS);
+        relationshipSet.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(relationshipSet);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        List<PropertyDescriptor> props = new ArrayList<>(properties);
+        props.add(FILE_OR_DIRECTORY);
+        props.add(RECURSIVE);
+        return props;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        String fileOrDirectoryName = null;
+        FlowFile flowFile = session.get();
+
+        // If this processor has an incoming connection, then do not run unless a
+        // FlowFile is actually sent through
+        if (flowFile == null && context.hasIncomingConnection()) {
+            context.yield();
+            return;
+        }
+
+        if (flowFile != null) {
+            fileOrDirectoryName = context.getProperty(FILE_OR_DIRECTORY).evaluateAttributeExpressions(flowFile).getValue();
+        } else {
+            fileOrDirectoryName = context.getProperty(FILE_OR_DIRECTORY).evaluateAttributeExpressions().getValue();
+        }
+
+        final FileSystem fileSystem = getFileSystem();
+        try {
+            // Check if the user has supplied a file or directory pattern
+            List<Path> pathList = Lists.newArrayList();
+            if (fileOrDirectoryName.contains("*")) {
```

Comment:
I did some research and it appears that there are some additional characters and patterns...I'll see if I can make a regex for recognizing some of the characters which would infer that a glob was provided. An another option is to have a true/false property which would explicitly specify that a glob is being provided.